packages/gestalt/src/PopoverEducational.js

```diff
@@ -0,0 +1,186 @@
+   */
+  primaryAction?: PrimaryActionType,
+  /**
+   * Main text content of PopoverEducational. Content should be [localized](https://gestalt.pinterest.systems/web/popovereducational#Localization). See the [message variant](https://gestalt.pinterest.systems/web/popovereducational#Message) to learn more.
+   *
+   */
+  message?: string | Element<typeof Text>,
+  /**
+   * The underlying ARIA role for PopoverEducational. See the [role sectionin Accessibility](https://gestalt.pinterest.systems/web/popovereducational#Role) for more info.
+   */
+  role?: Role,
+  /**
+   * The maximum width of PopoverEducational. See the [size variant](https://gestalt.pinterest.systems/web/popovereducational#Size) to learn more.
+   */
+  size?: Size,
+  /**
+   * An object representing the zIndex value of PopoverEducational. Learn more about [zIndex classes](https://gestalt.pinterest.systems/web/zindex_classes)
+   */
+  zIndex?: Indexable,
+|};
+
+/**
+ * [PopoverEducational](https://gestalt.pinterest.systems/web/popovereducationaleducational) is a floating container that introduces users to elements on the screen. Used for education or onboarding experiences.
+ * ![PopoverEducational light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/PopoverEducational.spec.mjs-snapshots/PopoverEducational-chromium-darwin.png)
+ * ![PopoverEducational dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/PopoverEducational-dark.spec.mjs-snapshots/PopoverEducational-dark-chromium-darwin.png)
+ */
+export default function PopoverEducational({
+  accessibilityLabel,
+  anchor,
+  children,
+  id,
+  idealDirection,
+  message,
+  onDismiss,
+  primaryAction,
+  role = 'tooltip',
+  size = 'sm',
+  zIndex,
+}: Props): Node {
+  if (!anchor) {
+    return null;
+  }
+
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  let textElement: Element<'span'> | Element<typeof Text>;
+
+  if (typeof message === 'string') {
+    textElement = <Text color="light">{message}</Text>;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  if (
+    message &&
+    typeof message !== 'string' &&
+    Children.only(message).type.displayName === 'Text'
+  ) {
+    let textColorOverrideStyles = isDarkMode
+      ? styles.textColorOverrideDark
+      : styles.textColorOverrideLight;
+
+    textElement = <span className={textColorOverrideStyles}>{message}</span>;
+  }
+
+  return (
+    <Box zIndex={zIndex}>
+      <Popover
+        accessibilityLabel={accessibilityLabel}
+        anchor={anchor}
+        color="blue"
+        id={id}
+        idealDirection={idealDirection}
+        onDismiss={onDismiss}
+        positionRelativeToAnchor
+        showCaret
+        role={primaryAction && !children ? 'dialog' : role}
+        size={size}
+      >
+        {message && !children ? (
```

Comment:
This logic seems a bit disconnected. I'd recommend something like
```js
{children ?? message ? (markup) : null}
```
That makes the relationship between `children` and `message` more clear

Reply:
DONE, good catch